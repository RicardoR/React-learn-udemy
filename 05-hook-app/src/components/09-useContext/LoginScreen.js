import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
  const { user, setUser } = useContext(UserContext);

  const handleLogin = () => {
    setUser({
      id: 123,
      name: 'John Doe',
    });
  };

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <pre>{JSON.stringify(user, null, 3)}</pre>

      <button className="btn btn-primary" onClick={() => handleLogin()}>
        Login
      </button>
    </div>
  );
};
