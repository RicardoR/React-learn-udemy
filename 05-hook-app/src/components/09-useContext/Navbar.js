import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import './styles.css';

export const Navbar = () => {
  return (
    <nav className="navbar navbar-expand-md navbar-dark bg-dark">
      <Link className="navbar-brand" to="#">
        useContext
      </Link>

      <button
        class="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span class="navbar-toggler-icon"></span>
      </button>
      <div className="collapse navbar-collapse" id="navbarNav">
        <ul className="navbar-nav">
          <li className="nav-item">
            <NavLink className="nav-link" to="/">
              Home
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink className="nav-link" to="/about">
              About
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink className="nav-link" to="/login">
              Login
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
  );
};
