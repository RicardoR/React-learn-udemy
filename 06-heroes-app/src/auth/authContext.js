import { createContext } from 'react';

// Uppercase because its a component
export const AuthContext = createContext();
