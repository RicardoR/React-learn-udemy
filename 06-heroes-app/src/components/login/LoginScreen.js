import { useNavigate } from 'react-router-dom';

export const LoginScreen = () => {
  const navigate = useNavigate();
  const handleLogin = () => {
    // replace true evita que puedas volver atr√°s a esta pantalla
    navigate('/', { replace: true });
  };

  return (
    <div className="container mt-5">
      <h1>LoginScreen</h1>
      <hr />
      <button className="btn btn-primary" onClick={handleLogin}>
        Login
      </button>
    </div>
  );
};
