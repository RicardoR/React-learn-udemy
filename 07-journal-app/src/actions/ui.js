import { actionTypes } from '../types/actionTypes';

export const setError = (message) => {
  return {
    type: actionTypes.UI_SET_ERROR,
    payload: message,
  };
};

export const removeError = () => {
  return {
    type: actionTypes.UI_REMOVE_ERROR,
  };
};
