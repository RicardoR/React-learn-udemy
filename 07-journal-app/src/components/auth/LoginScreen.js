import React, { useState } from 'react';
import validator from 'validator';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import {
  startLoginWithEmailPassword,
  startGoogleLogin,
} from '../../actions/auth';
import { useForm } from '../../hooks/useForm';

export const LoginScreen = () => {
  const dispatch = useDispatch();
  const [error, setError] = useState(undefined);
  const { msgError, loading } = useSelector((state) => state.ui);

  const [formValues, handleInputChange] = useForm({
    email: 'email@gmail.com',
    password: '123456',
  });

  const { email, password } = formValues;

  const handleLogin = (event) => {
    event.preventDefault();
    if (isFormValid()) {
      dispatch(startLoginWithEmailPassword(email, password));
    }
  };

  const handleGoogleLogin = () => {
    if (!loading) {
      dispatch(startGoogleLogin());
    }
  };

  const isFormValid = () => {
    if (!email || !password) {
      setError('All fields are required');
      return false;
    }

    if (!validator.isEmail(email)) {
      setError('Invalid email');
      return false;
    }

    if (password.length < 5) {
      setError('Password must be at least 5 characters');
      return false;
    }

    setError(undefined);
    return true;
  };

  return (
    <>
      <h3 className="auth__title mb-5">Login</h3>
      <form onSubmit={handleLogin}>
        {(error || msgError) && (
          <p className="alert-error mb-5">{error || msgError}</p>
        )}
        <input
          type="text"
          placeholder="Email"
          name="email"
          className="auth__input"
          value={email}
          onChange={handleInputChange}
        />
        <input
          type="password"
          placeholder="password"
          name="password"
          className="auth__input"
          autoComplete="off"
          value={password}
          onChange={handleInputChange}
        />
        <button
          disabled={loading}
          type="submit"
          className="btn btn-primary btn-block mb-5 mt-1"
        >
          Login
        </button>
        <div>
          <p className="mb-5">Login with social networks</p>

          <div className="google-btn mb-5" onClick={handleGoogleLogin}>
            <div className="google-icon-wrapper">
              <img
                className="google-icon"
                src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
                alt="google button"
              />
            </div>
            <p className="btn-text">
              <b>Sign in with google</b>
            </p>
          </div>
        </div>
        <Link className="link" to="/auth/register">
          Create new account
        </Link>
      </form>
    </>
  );
};
