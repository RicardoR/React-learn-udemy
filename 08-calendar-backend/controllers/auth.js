const { response } = require('express');

const createUser = (request, resp = response) => {
  const { name, email, password } = request.body;

  if (name.length < 5) {
    return resp.status(400).json({
      ok: false,
      msg: 'El nombre debe tener 5 caracteres',
    });
  }

  resp.status(201).json({
    ok: true,
    msg: 'registro',
    name,
    email,
    password,
  });
};

const performLogin = (request, resp = response) => {
  const { email, password } = request.body;

  resp.json({
    ok: true,
    msg: 'login',
    email,
    password,
  });
};

const renewToken = (request, resp = response) => {
  const { token } = request.body;

  resp.json({
    ok: true,
    msg: 'renew',
    token,
  });
};

module.exports = { createUser, performLogin, renewToken };
