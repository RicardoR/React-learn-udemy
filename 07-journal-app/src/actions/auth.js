import { getAuth, signInWithPopup } from 'firebase/auth';

import { actionTypes } from '../types/actionTypes';
import { googleAuthProvider } from '../firebase/firebase-config';

export const initLogin = (email, password) => {
  return (dispatch) => {
    setTimeout(() => {
      dispatch(login(123, 'pedro'));
    }, 3500);
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    const auth = getAuth();
    signInWithPopup(auth, googleAuthProvider).then(({ user }) => {
      dispatch(login(user.uid, user.displayName));
    });
  };
};

export const login = (uid, displayName) => {
  return {
    type: actionTypes.LOGIN,
    payload: {
      uid,
      displayName,
    },
  };
};
