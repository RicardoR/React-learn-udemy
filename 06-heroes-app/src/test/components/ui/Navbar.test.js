import { mount } from 'enzyme';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { AuthContext } from '../../../auth/authContext';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { Navbar } from '../../../components/ui/Navbar';
import { types } from '../../../types/types';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('Navbar Ut', () => {
  const contextValue = {
    user: {
      logged: true,
      name: 'Ricardo',
    },
    dispatch: jest.fn(),
  };
  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter initialEntries={['/']}>
        <Routes>
          <Route path="/" element={<Navbar />} />
          <Route path="/login" element={<LoginScreen />} />
        </Routes>
      </MemoryRouter>
    </AuthContext.Provider>
  );
  test('should display with default values', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.text-info').text().trim()).toBe('Ricardo');
  });

  test('should call logout, navigate and dispatch with args', () => {
    wrapper.find('button').simulate('click');
    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.logout,
    });
    expect(mockNavigate).toHaveBeenCalledWith('/login', { replace: true });
  });
});
