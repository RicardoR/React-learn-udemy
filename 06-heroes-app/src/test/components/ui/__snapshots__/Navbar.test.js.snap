// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Ut should display with default values 1`] = `ReactWrapper {}`;
